src/Beman/Optional26/tests/__utils.h

```diff
@@ -0,0 +1,36 @@
+#ifndef BEMAN_OPTIONAL26_OPTIONAL_TESTS__UTILS_HPP
```

Comment:
utils is a sure to collide name for things, and the names of types in here are, too. This should be beman/optional/test/utils.hpp or something similar? 
Namespace shouldn't be just ::test, or when this is composed into a larger group it's going to collide when linked, even if no TU ends up with two "::test::no_default_ctor" types.